anchore_engine/services/policy_engine/engine/vulns/cpe_matchers.py

```diff
@@ -0,0 +1,330 @@
+from typing import List, Tuple
+
+from sqlalchemy.orm import Session
+from collections import namedtuple
+
+from anchore_engine.db import (
+    NvdV2Metadata,
+    CpeV2Vulnerability,
+    Image,
+    get_thread_scoped_session,
+    ImageCpe,
+    VulnDBCpe,
+    Vulnerability,
+    DistroNamespace,
+    ImagePackage,
+)
+from anchore_engine.services.policy_engine.engine.vulns.cpes import (
+    FuzzyCandidateCpeGenerator,
+    dedup_cpe_vulnerabilities,
+)
+from anchore_engine.services.policy_engine.engine.vulns.db import (
+    CpeDBQueryManager,
+    db_result_tuples_to_list,
+)
+from anchore_engine.subsys import logger
+from anchore_engine.utils import timer
+
+# Match class to bind the image and vuln sides and ease comparisons and hashing
+CpeMatch = namedtuple("CpeMatch", ["image_cpe", "vuln_cpe"])
+
+
+def cve_ids_for_vuln_record(vuln: Vulnerability) -> List[str]:
+    """
+    Get CVE Id for vuln from alpine record
+
+    :param vuln:
+    :return: list of CVE ids associated with the vulnerability record
+    """
+
+    if vuln.id.startswith("CVE-"):
+        ids = [vuln.id]
+    else:
+        ids = []
+
+    # Some records have references in the metadata structure as (Amazon ALAS, for an example):
+    # {"CVE": ["CVE-1", "CVE-2",...]}
+    try:
+        refs = vuln.additional_metadata["CVE"]
+    except (KeyError, AttributeError, TypeError):
+        refs = []
+
+    ids.extend(refs)
+    if len(ids) > 1:
+        # Ensure no duplicates
+        return list(set(ids))
+    else:
+        return ids
+
+
+def filter_secdb_entries(
+    image_distro: DistroNamespace, matches: List[str], db_manager: CpeDBQueryManager
+) -> List[str]:
+    """
+    Execute the filtering functionality itself on the sets
+
+    :param image_distro:
+    :param matches: match list to filter
+    :return: filtered match list
+    """
+
+    secdb_matched_cves = db_manager.matched_records_for_namespace(image_distro, matches)
+
+    logger.spew("Secdb matched cves %s", secdb_matched_cves)
```

Comment:
side note: spew is something that my JSON logging PR removes in favor of debug. Not saying that was the right call or anything, but food for thought.

Reply:
Ack. I think we need a level of verbosity for development. We wouldn't ask a user to enable that log level due to the amount of data it outputs, but provides enough detail for dev usage. But duly noted and we can update consistently when we make the broader change.